src/imap.rs

```diff
@@ -1119,10 +1119,26 @@ impl Imap {
             .session
             .as_mut()
             .context("IMAP No Connection established")?;
-        let self_addr = context.get_configured_addr().await?;
-        let search_command = format!("FROM \"{}\"", self_addr);
+
+        // TODO probably requires a test if want to keep it as is:
+        let self_addrs = context
+            .get_all_self_addrs()
+            .await
+            .context("get_all_recipients() can't get self addrs")?;
+
+        if self_addrs.is_empty() {
+            bail!("get_all_recipients(): No self addresses");
+        }
+
+        let from_part = self_addrs
+            .iter()
+            .map(|a| format!(" FROM {}", a))
+            .collect::<String>();
+        let mut search_command = " OR".repeat(self_addrs.len() - 1);
+        search_command.push_str(&from_part);
+
```

Comment:
TODO either make an integration test for `get_all_recipients()` while there are multiple secondary addresses, or make a unit test just for `search_command` and test this manually once (I don't think I ever did).

Reply:
implemented second suggestion and fixed the imap search construct (but didn't test it functionally, maybe you can add/do that).